From d827d4850e405f098ed0967f02ef3034dbddd661 Mon Sep 17 00:00:00 2001
From: Xeonacid <h.dwwwwww@gmail.com>
Date: Wed, 19 Jan 2022 23:19:35 +0800
Subject: [PATCH] fix(changeUserPwd/run-passwd.cpp): format security

Add "%s" format to g_error_new, to fix the format string security issue.
---
 changeUserPwd/run-passwd.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/changeUserPwd/run-passwd.cpp b/changeUserPwd/run-passwd.cpp
index 59a76501..aca032f1 100644
--- a/changeUserPwd/run-passwd.cpp
+++ b/changeUserPwd/run-passwd.cpp
@@ -377,7 +377,7 @@ static gboolean io_watch_stdout (GIOChannel *source, GIOCondition condition, Pas
                                          "Your password has been changed after you verify!");
                 } */else {
                     error = g_error_new (PASSWD_ERROR, PASSWD_ERROR_UNKNOWN,
-                                         str->str);
+                                         "%s", str->str);
                 }
 
                 /* At this point, passwd might have exited, in which case
